spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/AbstractAzureMessageHandler.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.codec.CodecMessageConverter;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.handler.AbstractMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.converter.MessageConverter;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Base class of outbound adapter to publish to azure backed messaging service
+ *
+ * <p>
+ * It delegates real operation to {@link SendOperation} and
+ * converts the {@link Message} payload into specific type {@link D} accepted by
+ * Azure SDK. It supports synchronous and asynchronous sending.
+ *
+ * @author Warren Zhu
+ */
+public abstract class AbstractAzureMessageHandler<D> extends AbstractMessageHandler {
+    private static final long DEFAULT_SEND_TIMEOUT = 10000;
+    private final String destination;
+    private final SendOperation<D> sendOperation;
+    protected MessageConverter messageConverter;
+    private boolean sync = false;
+    private ListenableFutureCallback<Void> sendCallback;
+    private EvaluationContext evaluationContext;
+    private Expression sendTimeoutExpression = new ValueExpression<>(DEFAULT_SEND_TIMEOUT);
+    private Expression partitionKeyExpression;
+
+    public AbstractAzureMessageHandler(String destination, SendOperation<D> sendOperation) {
+        Assert.hasText(destination, "destination can't be null or empty");
+        Assert.notNull(sendOperation, "sendOperation can't be null");
```

Comment:
How about `@NonNull` from springframework ?

Reply:
i think we should be consistent across all our Java code base e.g. annotation, getter setter etc.

Reply:
Agree.